frontend/src/app/services/codelist/codelist.service.ts

```diff
@@ -93,48 +98,52 @@ export class CodelistService {
     language = "de",
     sortBy:
       | CodelistSort
-      | ((a: SelectOptionUi, b: SelectOptionUi) => number) = "label",
+      | ((a: CodelistEntry, b: CodelistEntry) => number) = "label",
   ): SelectOptionUi[] => {
     if (!codelist) {
       return [];
     }
+    // Sort codelist entries
+    const sortFunction =
+      typeof sortBy === "function"
+        ? sortBy
+        : CodelistService.getDefaultSortFunction(sortBy);
```

Comment:
Kann das funktionieren? sortBy bezog sich auf die gemappten Codelisten, also SelectOptionUi-Elemente. Ein CodelistEntry hat andere Felder.

Reply:
Das sortBy bezieht sich nun auf CodeListEntries, die Funktion getSortFunction dient nun zur Bestimmung der Felder.

Ich denke, das spreaden ist hier nicht wirklich nötig, aber die Tests schlugen ohne fehl. https://stackoverflow.com/questions/62662695/confused-console-log-is-outputting-a-sorted-array-before-the-sort-operates-on-it

const sortedEntries: CodelistEntry[] = [...codelist.entries].sort(
sortFunction,
);